pkg/binary/envoytest/util.go

```diff
@@ -115,24 +114,25 @@ func RequireRunTerminate(t *testing.T, r binary.Runner, options RunKillOptions)
 	}
 
 	// This ensures on any panic the envoy process is terminated, which can prevent test hangs.
-	deferredTerminate := func() {
-		// envoy.waitForTerminationSignals() registers SIGINT and SIGTERM
-		r.SendSignal(syscall.SIGTERM)
+	deferredInterrupt := func() {
+		r.FakeInterrupt()
 	}
 
 	defer func() {
-		if deferredTerminate != nil {
-			deferredTerminate()
+		if deferredInterrupt != nil {
```

Comment:
I don't see how `deferredInterrupt` can be `nil`. Maybe you just want to call it on defer directly? Also, unless `deferredInterrupt` gets mutated in the future put the if wrapping the defer to save some CPU :p

Reply:
because it drifted! great eye